File: C/test/test_findPoisonedDuration.c

#ifdef TEST

#include "unity.h"

#include "findPoisonedDuration.h"

void setUp(void)
{
}

void tearDown(void)
{
}

void test_findPoisonedDuration_Example_1(void)
{
	int input0[] = {1, 4};
	int input1 = 2;
	int expected = 4;
	TEST_ASSERT_EQUAL(expected, findPoisonedDuration(input0, sizeof(input0) / sizeof(input0[0]), input1));
}

void test_findPoisonedDuration_Example_2(void)
{
	int input0[] = {1, 2};
	int input1 = 2;
	int expected = 3;
	TEST_ASSERT_EQUAL(expected, findPoisonedDuration(input0, sizeof(input0) / sizeof(input0[0]), input1));
}

#endif   // TEST

File: Rust/tests/test_find_poisoned_duration.rs
use leetcode_rust::solution::Solution;

#[test]
fn test_example_1() {
    let input0 = vec![1, 4];
    let input1 = 2;
    let expected = 4;
    assert_eq!(Solution::find_poisoned_duration(input0, input1), expected);
}

#[test]
fn test_example_2() {
    let input0 = vec![1, 2];
    let input1 = 2;
    let expected = 3;
    assert_eq!(Solution::find_poisoned_duration(input0, input1), expected);
}
